identity-service/src/notifications/processNotifications/supporterRankChangeNotification.js

```diff
@@ -0,0 +1,41 @@
+
+const models = require('../../models')
+const { notificationTypes } = require('../constants')
+
+async function processSupporterRankChangeNotification (notifications, tx) {
+  for (const notification of notifications) {
+    const { slot, initiator: receiverUserId, metadata: { entity_id: senderUserId, rank } } = notification
+
+    await Promise.all([
```

Comment:
would we maybe wanna do a Promise.all on the loop itself, and also just checking that its necessary for promises to resolve before returning notifs? also no worries if the timing of this stuff is all v short, just curious!

Reply:
This is a super good point. I was following the existing pattern, but would be much faster to Promise.all outside of the loop